node-wasm/src/utils.rs

```diff
@@ -71,27 +73,10 @@ pub(crate) fn js_value_from_display<D: fmt::Display>(value: D) -> JsValue {
     JsValue::from(value.to_string())
 }
 
-pub(crate) trait JsContext<T> {
-    fn js_context<C>(self, context: C) -> Result<T, JsError>
-    where
-        C: fmt::Display + Send + Sync + 'static;
-}
-
-impl<T, E> JsContext<T> for std::result::Result<T, E>
-where
-    E: std::error::Error,
-{
-    fn js_context<C>(self, context: C) -> Result<T, JsError>
-    where
-        C: fmt::Display + Send + Sync + 'static,
-    {
-        self.map_err(|e| JsError::new(&format!("{context}: {e}")))
-    }
-}
-
-pub(crate) fn get_crypto() -> Result<Crypto, JsError> {
+pub(crate) fn get_crypto() -> Result<Crypto, Error> {
     js_sys::Reflect::get(&js_sys::global(), &JsValue::from_str("crypto"))
```

Comment:
by the looks of it, it should work both in normal js and in worker, nice one!

Reply:
Yes. That is why I did it like this.